src/main/java/nl/igorski/example/MWEngineActivity.java

```diff
@@ -255,6 +261,10 @@ protected void setupSong()
 
     protected void flushSong()
     {
+        // this ensures that Song resources currently in use by the engine are released
+
+        _engine.pause();
+
```

Comment:
This should ensure that when we dispose all of the songs allocated resources they aren't in use by the engine during a rendering cycle.